src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1142,9 +1149,20 @@
   (time! (:store-report metrics)
          (let [{:keys [puppet_version certname report_format configuration_version
                        start_time end_time resource_events transaction_uuid environment
-                       status noop]
-                :as report}         (normalize-report orig-report)
-                report-hash         (shash/report-identity-hash report)]
+                       status noop] :as report}         (normalize-report orig-report)
+                report-hash         (shash/report-identity-hash report)
+        containment-path-fn (fn [cp] (if-not (nil? cp) (sutils/to-jdbc-varchar-array cp)))
+        report-metrics (if (scf-utils/postgres?)
+                         (scf-utils/clj->pgobject (:metrics report))
```

Comment:
metrics is a global var in this file, so that won't work without changing the name of the metrics variable. I don't care particularly though, happy to make the change.